server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterHealthIndicatorService.java

```diff
@@ -155,7 +157,14 @@ private HealthIndicatorDetails getDetails(
                 });
             }
             if (coordinationDiagnosticsDetails.nodeToClusterFormationDescriptionMap() != null) {
-                builder.field(CLUSTER_FORMATION, coordinationDiagnosticsDetails.nodeToClusterFormationDescriptionMap());
+                builder.field(
+                    CLUSTER_FORMATION,
+                    coordinationDiagnosticsDetails.nodeToClusterFormationDescriptionMap()
+                        .entrySet()
+                        .stream()
+                        .map(entry -> Map.of("node_id", entry.getKey(), CLUSTER_FORMATION_MESSAGE, entry.getValue()))
```

Comment:
In light of https://github.com/elastic/elasticsearch/issues/89664 shall we use a different key instead of `node_id` ? 
Maybe `node` ?

Or otherwise, maybe report `node_id` and `node_name` moving forward as separate keys? (similar to how we do it in `recent_masters` ?

What do you think? // cc @dakrone

Reply:
Since it's no longer the key, it made sense to me to explicitly put the node_id here. I'm planning on a follow up PR (or PRs) to deal with adding the name, but wanted to keep the scope of this one small.

Reply:
Ah ++ 
Maybe we can tag https://github.com/elastic/elasticsearch/issues/89664 in the follow up PR and change the description to reporting both id and name (which I like better than combining the 2)